rust/rope/src/rope.rs

```diff
@@ -443,7 +459,7 @@ impl<'de> Deserialize<'de> for Delta<RopeInfo> {
         #[serde(rename_all = "snake_case")]
         enum RopeDeltaElement_ {
             Copy(usize, usize),
-            Insert(String),
+            Insert(Node<RopeInfo>),
```

Comment:
hmm, I wonder if we can get rid of this whole custom deserialize impl now? The whole strategy of deserializing to internal types and then converting might not be necessary now that we have the visitor? I'm not totally sure if this is the case, since I forget exactly what the constraints were that inspired this originally, but it was something like this. Maybe worth playing around with for a few minutes?

Reply:
If we want to derive Serialize and Deserialize we would have to add those traits to the NodeInfo trait in DeltaElement and Delta, since we can't use derive for a concrete case like Delta<RopeInfo>

Reply:
Okay, thanks, let's keep it as is then.